### PR TITLE
*: add a tidb system variable tidb_hash_join_concurrency

### PR DESCRIPTION
**Pull request by XuHuaiyu**
This commit adds a tidb system variable called tidb_hash_join_concurrency,
thus we can change the value of hash join concurrency without restarting the tidb server.

PTAL @zz-jason @winoros

### CODE REVIEW

File: sessionctx/variable/tidb_vars.go [hunk 1]
```diff
@@ -138,6 +100,47 @@ const (
 	TiDBEnableStreaming = "tidb_enable_streaming"
 )
 
+// TiDB session and global system variable names.
```
Comment (jackysp) [inline sessionctx/variable/tidb_vars.go]: `session or global` is better.

File: sessionctx/variable/session.go [hunk 2]
```diff
@@ -406,6 +410,15 @@ func (s *SessionVars) GetStatusFlag(flag uint16) bool {
 	return s.Status&flag > 0
 }
 
+// GetHashJoinConcurrency gets the value of SessionVars.HashJoinConcurrency.
+func (s *SessionVars) GetHashJoinConcurrency() uint {
```
Comment (zz-jason) [inline sessionctx/variable/session.go]: can we remove this func and initialize `HashJoinConcurrency` in other places, thus we can use `HashJoinConcurrency` like other session variables ?

**Comment by shenli** [comment]
Please resolve the conflicts and fix CI.

**Comment by XuHuaiyu** [comment]
/run-integration-common-test tidb-test=pr/496 tidb-private-test=pr/8
/run-common-test tidb-test=pr/496 tidb-private-test=pr/8

**Comment by XuHuaiyu** [comment]
PTAL @zz-jason @jackysp

**Comment by XuHuaiyu** [comment]
/run-integration-common-test tidb-test=pr/496 tidb-private-test=pr/8

**Pull request merged**